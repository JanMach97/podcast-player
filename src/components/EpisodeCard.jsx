import React, { Component } from "react";
import * as api from "../api";

class EpisodeCard extends Component {
  state = {
    episode: {}
  };
  render() {
    return (
      <div className="episode-card">
        <header>
          <img src={this.state.episode.image_url} alt="" />
          {Object.keys(this.state.episode).length !== 0 ? (
            <div className="header-right">
              <h2>{this.state.episode.title}</h2>
              <p>Author: {this.state.episode.author.fullname}</p>
              <p>Pusblished at: {this.state.episode.published_at}</p>
            </div>
          ) : (
            ""
          )}
        </header>
        <div className="description">
          <p>{this.state.episode.description}</p>
        </div>
      </div>
    );
  }

  componentDidMount() {
    api
      .fetchSingleEpisode(this.props.match.params.episode_id)
      .then(({ response: { episode } }) => {
        this.setState({
          episode
        });
      });
  }
}

export default EpisodeCard;
