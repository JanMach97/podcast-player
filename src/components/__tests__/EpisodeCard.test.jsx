import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import EpisodeCard from '../EpisodeCard'

configure({ adapter: new Adapter() });

describe('EpisodeCard', () => {
  it('renders without crashing', () => {
    shallow(<EpisodeCard match={{ params: { episode_id: '16105972'}}}/>)
  })
})