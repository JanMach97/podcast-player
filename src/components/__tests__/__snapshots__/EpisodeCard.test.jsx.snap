// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCard renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EpisodeCard
    match={
      Object {
        "params": Object {
          "episode_id": "16105972",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <img
            alt=""
          />
          
        </header>,
        <div
          className="description"
        >
          <p />
        </div>,
      ],
      "className": "episode-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
            />,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "",
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p />,
          "className": "description",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <img
              alt=""
            />
            
          </header>,
          <div
            className="description"
          >
            <p />
          </div>,
        ],
        "className": "episode-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
              />,
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "",
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
